src/test/java/com/petclinic/selenium/seleniumtest/VISE_SeleniumTests.java

```diff
@@ -0,0 +1,135 @@
+        System.setProperty("sel.jup.output.folder", "./src/test/onFailureScreenshots");
+    }
+
+    public static void takeSnapShot(WebDriver webDriver, String fileWithPath) throws Exception {
+
+        //Convert web driver object to TakeScreenshot
+        TakesScreenshot scrShot = ((TakesScreenshot) webDriver);
+        //call getScreenshotAs method to create the actual image file
+        File SrcFile = scrShot.getScreenshotAs(OutputType.FILE);
+        //Move image file to new destination
+        File DestFile = new File(fileWithPath);
+        //Copy file at destination
+        FileUtils.copyFile(SrcFile, DestFile);
+    }
+    @Test
+    @DisplayName("test_create_a_visit")
+    void test_create_a_visit(TestInfo testInfo) throws Exception {
+        driver.get("http://localhost:8080");
+        driver.manage().window().maximize();
+
+        //enter authentication information
+        driver.findElement(By.id("email")).sendKeys("admin");
+        driver.findElement(By.id("pwd")).sendKeys("admin");
+        driver.findElement(By.id("button")).click();
+
+
+        TimeUnit.SECONDS.sleep(5);
+
+
+
+
+
+        //navigation to visits page
+        driver.findElement(By.linkText("Owners")).click();
+        TimeUnit.SECONDS.sleep(2);
+        WebElement a1 = driver.findElement(By.xpath("//*[@id=\"navbarSupportedContent\"]/ul[1]/li[4]/div/a[1]"));
+        a1.click();
+        TimeUnit.SECONDS.sleep(2);; // VERY VERY VERY NECESSARY PART OF THE CODE. Without the line it will try to click on something that doesn't exist yet.
+        a1 = driver.findElement(By.linkText("Betty Davis"));
+        a1.click();
+        TimeUnit.SECONDS.sleep(2);
+        a1 = driver.findElement(By.linkText("Add Visit"));
+        a1.click();
+
+
+
+        //getting a vet
+        //driver.findElement(By.id("selectedVet")).click();
+
+
+
+        //Selecting the vet : "Helen"
+        TimeUnit.SECONDS.sleep(2);
+        driver.findElement(By.xpath("//*[@id=\"selectedVet\"]/option[1]")).click();
+        TimeUnit.SECONDS.sleep(2);
+        String selectedVet = driver.findElement(By.xpath("//*[@id=\"selectedVet\"]/option[3]")).getText();
+        System.out.println(selectedVet +" : vetname");
+        assertThat(selectedVet, is("Helen Leary"));
+        // Necessary scroll
+        JavascriptExecutor js = (JavascriptExecutor) driver;
+        js.executeScript("window.scrollBy(0,250)", "");
+
+        driver.findElement(By.xpath("//*[@id=\"selectedVet\"]/option[3]")).click();
+        TimeUnit.SECONDS.sleep(2);
+
+
+        //typing TestVisitDescription inside of the text box labelled "description"
+
+        driver.findElement(By.xpath("//*[@id=\'description_textarea\']")).sendKeys("TestVisitDescription");
+
+        TimeUnit.SECONDS.sleep(2);
+
+       driver.findElement(By.id("submit_button")).click();
+
+
+
+
+        // getting back to the betty davis customer
+        // CHANGE IN THE FUTURE
+
+        Thread.sleep(2000);
```

Comment:
I know this has already been mentioned but you should use waits instead of Thread.sleep. Here is an article about the differences between the two: [https://www.lambdatest.com/blog/threadsleep-java-selenium/#:~:text=sleep()%20in%20Selenium%20Java%20because%20it%20is%20a%20static,times%20in%20our%20automation%20scripts]()

Reply:
Thank you for your comment. I have used wait and the test is now 10x faster .